File: src/main/java/com/github/joelws/release/tracker/service/release/execution/DeleteReleaseServiceExecution.java
package com.github.joelws.release.tracker.service.release.execution;

import com.github.joelws.release.tracker.dao.release.ReleaseDao;
import com.github.joelws.release.tracker.service.ServiceExecution;

public class DeleteReleaseServiceExecution implements ServiceExecution<String, Void> {
    private final ReleaseDao releaseDao;

    public DeleteReleaseServiceExecution(ReleaseDao releaseDao) {
        this.releaseDao = releaseDao;
    }

    @Override
    public Void execute(String identifier) {
        releaseDao.delete(identifier);
        return null;
    }
}

File: src/main/kotlin/com/github/joelws/release/tracker/service/release/execution/DeleteReleaseServiceExecution.kt
package com.github.joelws.release.tracker.service.release.execution

import com.github.joelws.release.tracker.dao.release.ReleaseDao
import com.github.joelws.release.tracker.service.ServiceExecution

open class DeleteReleaseServiceExecution(private val releaseDao: ReleaseDao) : ServiceExecution<String?, Unit> {

    override fun execute(param: String?) {
        releaseDao.read(param)?.let {
            releaseDao.delete(param)
        }
    }
}
